File: src/test/java/com/fluxchess/pulse/FileTest.java
/*
 * Copyright (C) 2013-2019 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class FileTest {

	@Test
	void testValues() {
		for (int file : File.values) {
			assertThat(File.values[file]).isEqualTo(file);
		}
	}
}

File: src/test/cpp/filetest.cpp
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include "file.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(filetest, testValues) {
  for (auto file : File::values) {
    EXPECT_EQ(file, File::values[file]);
  }
}

TEST(filetest, testIsValid) {
  for (auto file : File::values) {
    EXPECT_TRUE(File::isValid(file));
  }

  EXPECT_FALSE(File::isValid(File::NOFILE));
}
